src/Workspaces/VisualBasicTest/CodeGeneration/AddImportsTests.vb

```diff
@@ -0,0 +1,301 @@
+﻿Imports Microsoft.CodeAnalysis.Editing
+Imports Microsoft.CodeAnalysis.Formatting
+Imports Microsoft.CodeAnalysis.Options
+Imports Microsoft.CodeAnalysis.Simplification
+Imports Xunit
+
+Namespace Microsoft.CodeAnalysis.VisualBasic.UnitTests.Editting
+    Public Class AddImportsTests
+        Private ReadOnly _ws As AdhocWorkspace = New AdhocWorkspace()
+        Private ReadOnly _emptyProject As Project
+
+        Public Sub New()
+            _emptyProject = _ws.AddProject(
+                ProjectInfo.Create(
+                    ProjectId.CreateNewId(),
+                    VersionStamp.Default,
+                    "test",
+                    "test.dll",
+                    LanguageNames.VisualBasic,
+                    metadataReferences:={TestReferences.NetFx.v4_0_30319.mscorlib}))
+        End Sub
+
+        Private Function GetDocument(code As String, Optional globalImports As String() = Nothing) As Document
+            code = code.Replace(vbLf, vbCrLf)
+
+            Dim project = _emptyProject
+
+            If globalImports IsNot Nothing Then
+                Dim gi = GlobalImport.Parse(globalImports)
+                project = project.WithCompilationOptions(DirectCast(project.CompilationOptions, VisualBasicCompilationOptions).WithGlobalImports(gi))
+            End If
+
+            Return project.AddDocument("test.cs", code)
+        End Function
+
+        Private Sub Test(initialText As String, importsAddedText As String, simplifiedText As String, Optional options As OptionSet = Nothing, Optional globalImports As String() = Nothing)
+
+            Dim doc = GetDocument(initialText, globalImports)
+            options = If(options, doc.Project.Solution.Workspace.Options)
+
+            Dim imported = ImportAdder.AddImportsAsync(doc, options).Result
+
+            If importsAddedText IsNot Nothing Then
+                importsAddedText = importsAddedText.Replace(vbLf, vbCrLf)
+                Dim formatted = Formatter.FormatAsync(imported, SyntaxAnnotation.ElasticAnnotation, options).Result
+                Dim actualText = formatted.GetTextAsync().Result.ToString()
+                Assert.Equal(importsAddedText, actualText)
+            End If
+
+            If simplifiedText IsNot Nothing Then
+                simplifiedText = simplifiedText.Replace(vbLf, vbCrLf)
+                Dim reduced = Simplifier.ReduceAsync(imported, options).Result
+                Dim formatted = Formatter.FormatAsync(reduced, SyntaxAnnotation.ElasticAnnotation, options).Result
+                Dim actualText = formatted.GetTextAsync().Result.ToString()
+                Assert.Equal(simplifiedText, actualText)
+            End If
+        End Sub
+
+        <Fact>
+        Public Sub TestAddImport()
+            Test(
+<x>Class C
+    Public F As System.Collections.Generic.List(Of Integer)
+End Class</x>.Value,
```

Comment:
Multiline strings could be used instead of `<x>...</x>.Value`, e.g.  
```
"Class C
    Public F As System.Collections.Generic.List(Of Integer)
End Class"
```